src/Bubble.vala

```diff
@@ -119,7 +119,25 @@ public class Notifications.Bubble : AbstractBubble {
             var image_overlay = new Gtk.Overlay ();
             image_overlay.valign = Gtk.Align.START;
 
-            if (notification.image_path != null) {
+            if (notification.image_data != null) {
+                try {
+                    var scale = get_style_context ().get_scale ();
+                    var pixbuf = notification.image_data.to_pixbuf_at_size (48 * scale, 48 * scale);
```

Comment:
Looks like the `MaskedImage` class already scales the input pixbuf, so probably no need to also have scaling in this method.

Reply:
Good catch, thanks!